dask/typing.py

```diff
@@ -2,12 +2,14 @@
 
 import abc
 from collections.abc import Callable, Hashable, Mapping, Sequence
-from typing import Any, Protocol, TypeVar, runtime_checkable
-
-try:
-    from IPython.display import DisplayObject
-except ImportError:
-    DisplayObject = Any
+from typing import TYPE_CHECKING, Any, Protocol, TypeVar, runtime_checkable
+
+if TYPE_CHECKING:
+    # IPython import is relatively slow. Avoid if not necessary
+    try:
+        from IPython.display import DisplayObject
+    except ImportError:
+        DisplayObject = Any
```

Comment:
We shouldn't handle ImportError inside TYPE_CHECKING. mypy will ignore missing imports and continue.